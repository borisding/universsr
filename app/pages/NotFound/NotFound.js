import React from 'react';
import { PageWrapper } from '@layout';
import { HttpStatus } from '@common/components';

const NotFound = () => (
  <PageWrapper title="Not Found">
    <HttpStatus statusCode={404}>
      <h3>404 - Page Not Found.</h3>
    </HttpStatus>
  </PageWrapper>
);

export default NotFound;
