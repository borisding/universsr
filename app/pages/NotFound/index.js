import React from 'react';
import { PageContainer, HttpStatus } from '../../components';

function NotFound() {
  return (
    <PageContainer title="Not Found">
      <HttpStatus statusCode={404}>
        <h3>404 - Page Not Found.</h3>
      </HttpStatus>
    </PageContainer>
  );
}

export default React.memo(NotFound);
