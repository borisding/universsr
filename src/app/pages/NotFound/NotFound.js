import React from 'react';
import { HttpStatus, PageContainer } from '@common/components';

const NotFound = () => (
  <PageContainer title="Not Found">
    <HttpStatus statusCode={404}>
      <h3>404 - Page Not Found.</h3>
    </HttpStatus>
  </PageContainer>
);

export default NotFound;
