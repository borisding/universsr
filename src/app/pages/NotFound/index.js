import React from 'react';
import { HttpStatus, PageTitle } from '@common/components';

const NotFound = () => (
  <PageTitle title="Not Found">
    <HttpStatus statusCode={404}>
      <h3>404 - Page Not Found.</h3>
    </HttpStatus>
  </PageTitle>
);

export default NotFound;
