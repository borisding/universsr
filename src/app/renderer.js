import React from 'react';
import DocumentTitle from 'react-document-title';
import serialize from 'serialize-javascript';
import flushChunks from 'webpack-flush-chunks';
import { renderToString } from 'react-dom/server';
import { flushChunkNames } from 'react-universal-component/server';
import { matchRoutes, renderRoutes } from 'react-router-config';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import storeFactory from './store';
import routes from './routes';

// preload data for matched route
function prefetchBranchData(store, url) {
  try {
    const branch = matchRoutes(routes, url);
    const promises = branch.map(({ route, match }) => {
      const { loadData } = route;
      const { dispatch } = store;

      if (match && match.isExact && loadData) {
        return Array.isArray(loadData)
          ? Promise.all(loadData.map(action => dispatch(action(match))))
          : dispatch(loadData(match));
      }

      return Promise.resolve(null);
    });

    return Promise.all(promises);
  } catch (err) {
    throw err;
  }
}

// export default server renderer and receiving stats
// also, should allow it to be mounted as middleware for production usage
export default function serverRenderer({ clientStats }) {
  return async (req, res, next) => {
    try {
      const context = {};
      const store = storeFactory();

      await prefetchBranchData(store, req.url);

      const appString = renderToString(
        <Provider store={store}>
          <StaticRouter location={req.url} context={context}>
            {renderRoutes(routes)}
          </StaticRouter>
        </Provider>
      );

      const pageTitle = DocumentTitle.rewind();
      const chunksOptions = { chunkNames: flushChunkNames() };
      const { scripts, stylesheets } = flushChunks(clientStats, chunksOptions);

      const preloadedState = serialize(store.getState(), { isJSON: true });
      const { statusCode = 200, redirectUrl } = context;
      const { nonce } = res.locals;

      if ([301, 302].includes(statusCode) && redirectUrl) {
        return res.redirect(statusCode, redirectUrl);
      }

      res.status(statusCode);
      res.render('index', {
        pageTitle,
        appString,
        preloadedState,
        stylesheets,
        scripts,
        nonce
      });
    } catch (err) {
      next(new Error(err));
    }
  };
}
