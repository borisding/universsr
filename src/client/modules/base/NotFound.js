import React from 'react';
import PageTitle from './PageTitle';
import Status from './Status';

const NotFound = () => {
  return (
    <PageTitle title="Not Found">
      <Status statusCode={404}>
        <h3>{'404 - Page Not Found.'}</h3>
      </Status>
    </PageTitle>
  );
};

export default NotFound;
