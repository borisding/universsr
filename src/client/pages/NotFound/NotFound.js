import React from 'react';
import { Container, HttpStatus } from '@client/common/components';

export default function NotFound() {
  return (
    <Container title="Not Found">
      <HttpStatus statusCode={404}>
        <h3>404 - Page Not Found.</h3>
      </HttpStatus>
    </Container>
  );
}
