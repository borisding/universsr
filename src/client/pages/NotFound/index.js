import lazy from '@client/pages/lazy';

export default lazy(() =>
  import(/* webpackChunkName: 'not-found' */ './NotFound')
);
