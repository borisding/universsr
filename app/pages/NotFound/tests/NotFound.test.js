import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NotFound from '../NotFound';

describe('NotFound', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NotFound />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
