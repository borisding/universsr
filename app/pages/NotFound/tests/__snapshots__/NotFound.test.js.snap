// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<h3>
  404 - Page Not Found.
</h3>
`;
